### PR TITLE
Fix for OpenAPI documentation for 401 and 403 responses being overridden when using @RolesAllowed

### LINKED ISSUE #23534: RolesAllowed overrides OpenAPI definitions for 401 and 403 responses
### Describe the bug

I noticed the when I added `APIResponse` annotations to my REST endpoint to document the 401 and 403 responses, the OpenAPI documentation overrides my descriptions, and instead shows "Not Authorized" (401) and "Not Allowed" (403).  I've traced this down to using the `RolesAllowed` annotation.  If I use `Authenticated`, or no annotation at all, then the documentation reflects the `APIResponse` annotations.

### Expected behavior

OpenAPI documentation should be using the provided `APIResponse` annotations to document the 401 and 403 endpoint responses.

### Actual behavior

OpenAPI documentation is using hard-coded "Not Authorized" and "Not Allowed" response descriptions for the 401 and 403 response status codes.  This appears to be coming from https://github.com/quarkusio/quarkus/blob/e21e338d19ef3d15e2eec07cd2fd613a76375565/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/filter/AutoRolesAllowedFilter.java#L80

### How to Reproduce?

See attached reproducer.  

1. Run in dev mode (./mvnw clean compile quarkus:dev)
2. check the OpenAPI documentation at http://localhost:8080/q/openapi.json
3. Note that the 401 and 403 documentation use the references
4. Change the GreetingResource class to use the RolesAllowed annotation (either at the class or method level, doesn't matter)
5. refresh the OpenAPI documentation
6. Note that the 401 and 403 documentation is now ignoring the references, and is instead hard-coded to "Not Authorized" and "Not Allowed"

### Output of `uname -a` or `ver`

Linux laverne 5.11.0-40-generic #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.13" 2021-10-19

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by nderwin**
Fix #23534
Signed-off-by:Nathan Erwin <nathan.d.erwin@gmail.com>

### CODE REVIEW

File: extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java [hunk 1]
```diff
@@ -22,6 +24,10 @@ public String secureEndpointWithoutSecurityAnnotation() {
         return "secret";
     }
 
+    @APIResponses({
+            @APIResponse(responseCode = "401", description = "Who are you?"),
+            @APIResponse(responseCode = "403", description = "You cannot do that.")
+    })
```
Comment (phillip-kruger) [inline extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java:30]: Rather add a new test method so we still have a test that test the auto creation of APIResponse in the case it's not provided and another that test the case that do not create them (like you have now)

File: extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java [hunk 2]
```diff
@@ -37,6 +43,10 @@ public String secureEndpoint1() {
...
```
Comment (phillip-kruger) [inline extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java:49]: same comment as above

**Review by phillip-kruger — CHANGES_REQUESTED** [review]
LGTM, just one comment. Let me know what you think

File: extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java (hunk 2 above)
Comment (nderwin) [reply extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java:49]: Ok, sounds good;  I'll try to add those later today.

Comment (nderwin) [reply extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiResourceSecuredAtMethodLevel.java]: @phillip-kruger I've updated the PR with separated out REST endpoints that are annotated to document the 401 and 403 responses, and a separate test method that just tests the 401 and 403 documentation.

**Review by phillip-kruger — APPROVED** [review]
Lgtm

**Comment by phillip-kruger** [comment]
Fix #23534

**Pull request merged**